sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java

```diff
@@ -0,0 +1,619 @@
+        return withContext(context -> updateQueueWithResponse(queue, context));
+    }
+
+    /**
+     * Package-private method that creates a queue with its context.
+     *
+     * @param queue Queue to create.
+     * @param context Context to pass into request.
+     *
+     * @return A Mono that completes with the created {@link QueueDescription}.
+     */
+    Mono<Response<QueueDescription>> createQueueWithResponse(QueueDescription queue, Context context) {
+        if (queue == null) {
+            return monoError(logger, new NullPointerException("'queue' cannot be null"));
+        } else if (queue.getName() == null || queue.getName().isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'queue.getName' cannot be null or empty."));
+        } else if (context == null) {
+            return monoError(logger, new NullPointerException("'context' cannot be null."));
+        }
+
+        final CreateQueueBodyContent content = new CreateQueueBodyContent()
+            .setType("application/xml")
+            .setQueueDescription(queue);
+        final CreateQueueBody createEntity = new CreateQueueBody()
+            .setContent(content);
+
+        context.addData(AZ_TRACING_NAMESPACE_KEY, SERVICE_BUS_TRACING_NAMESPACE_VALUE);
+
+        try {
+            return queuesClient.putWithResponseAsync(queue.getName(), createEntity, null, context)
+                .map(response -> deserializeQueue(response, queue.getName()));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Package-private method that deletes a queue with its context.
+     *
+     * @param queueName Name of queue to delete.
+     * @param context Context to pass into request.
+     *
+     * @return A Mono that completes with the created {@link QueueDescription}.
+     */
+    Mono<Response<Void>> deleteQueueWithResponse(String queueName, Context context) {
+        if (queueName == null) {
+            return monoError(logger, new NullPointerException("'queueName' cannot be null"));
+        } else if (queueName.isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'queueName' cannot be empty."));
+        } else if (context == null) {
+            return monoError(logger, new NullPointerException("'context' cannot be null."));
+        }
+
+        context.addData(AZ_TRACING_NAMESPACE_KEY, SERVICE_BUS_TRACING_NAMESPACE_VALUE);
+
+        try {
+            return queuesClient.deleteWithResponseAsync(queueName, context)
+                .map(response -> {
+                    return new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
+                        response.getHeaders(), null);
+                });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Package-private method that gets a queue with its context.
```

Comment:
`Package-private method ` : I have never seen that writing in java doc, it is evident from the fact that there is no qualifier. 
I am fine with this but it is not usual.